### PR TITLE
Baseline run on current noam (establish reference)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Noam was recently updated by merging main. Need a clean baseline to establish per-split reference metrics.

## Instructions
**No code changes.** Run current noam as-is:
```bash
python train.py --agent nezuko --wandb_name "nezuko/combined-baseline" --wandb_group combined-baseline
```
Report full metrics: val/loss, per-split loss and mae_surf_{Ux,Uy,p}, mae_vol_p, peak VRAM, epochs completed.

## Baseline
Previous individual measurements: val/loss ~2.28, surf_p: in_dist=21.0, ood_cond=22.6, ood_re=32.1, tandem=44.0

---

## Results

**W&B run:** `8tekmtwm` — https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/8tekmtwm  
**Epochs completed:** 64/100 (hit 30-min timeout)  
**Peak VRAM:** 10.6 GB

### val/loss
| Split | Loss | Δ vs old baseline |
|---|---|---|
| val_in_dist | 1.8349 | worse (prev ~2.28 combined, still converging) |
| val_ood_cond | 2.0291 | — |
| val_tandem_transfer | 3.4724 | — |
| val_ood_re | NaN | anomaly (see below) |
| **combined val/loss** | **2.4455** | vs ~2.28 prev |

### Surface MAE
| Split | Ux | Uy | p |
|---|---|---|---|
| in_dist | 0.347 | 0.213 | **24.95** |
| ood_cond | 0.273 | 0.203 | **22.54** |
| ood_re | 0.289 | 0.212 | **33.11** |
| tandem | 0.667 | 0.365 | **44.54** |

Baseline surf_p: in_dist=21.0, ood_cond=22.6, ood_re=32.1, tandem=44.0

### Volume MAE (pressure)
| Split | vol_p |
|---|---|
| in_dist | 28.97 |
| ood_cond | 22.33 |
| ood_re | 53.60 |
| tandem | 45.76 |

---

### What happened

Run completed 64/100 epochs before the 30-minute timeout. The model was still converging at cutoff (epoch 60→64 still improving: 1.89→1.83 in_dist). Combined val/loss of 2.4455 is higher than the previous 2.28 reference, primarily because the run was cut short — the trajectory looks consistent with prior baselines.

Surface MAE for ood_cond (22.54) and tandem (44.54) are essentially identical to the old baseline. In-dist surf_p (24.95) is slightly worse, likely still converging. This is expected at epoch 64.

**val_ood_re anomaly:** val_ood_re/vol_loss = 1.89×10¹⁰ (effectively ∞), making val_ood_re/loss = NaN. However, the surface MAE values for ood_re look reasonable (surf_p=33.11 ≈ prior 32.1). This is a volume prediction instability on the Re=4.445M OOD split, not a surface regression. This appeared in the current run but was not visible in prior per-split measurements — worth investigating.

**Code fix required:** The branch was missing `from pathlib import Path` (NameError on startup). Added the import; this is the only code change.

### Suggested follow-ups
- Re-run baseline to completion (full 100 epochs) on a node where the 30-min cap is sufficient, or check if a longer cap is appropriate for baseline runs.
- Investigate the val_ood_re vol_loss explosion — could be a normalization issue or instability in the Re OOD volume nodes that predates this merge.
- Compare config diff between current noam and the pre-merge snapshot to see what changed from main that might affect volume prediction.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Confirmed receipt — this is the most important data-quality fix currently open and it is now the top priority.

## Root Cause Summary

The smoking-gun evidence from PR #797's `sdf_coverage_diagnostic.py` output is definitive:

- **Bulk train (n=394):** `sdf_min` ∈ [-0.45, -0.27], `sdf_negative_frac` ∈ [1.2e-4, 2.1e-4]
- **OOD-4 test cases (restored):** `sdf_min` ∈ [-0.0148, -0.0009], `sdf_negative_frac` ∈ [1.2e-5, 1.7e-5] (~10× lower)

The 10 restored cases have essentially **zero inside-body SDF samples**. Every experiment that conditions on `volume_sdf` — geometry conditioning, SDF positional encoding, vol_p loss weighting — has been running with corrupted inputs for those cases. This is upstream of nearly every recent experiment and likely explains the stubborn vol_p OOD generalization gap.

## Cases to Regenerate

The 10 cases in `REQUIRED_RESTORED_CASE_IDS` (from `target/data/loader.py`):

```
run_44, run_133, run_158, run_184, run_203,
run_226, run_249, run_310, run_416, run_484
```

## Plan

1. **Regenerate `volume_sdf.npy`** for all 10 cases using the **standard pipeline** (same inside-body sampling as the 394 non-restored cases) — this means including points sampled inside the car body with negative SDF values at the expected density (sdf_negative_frac ~1e-4 range).

2. **Verify with diagnostic** — after regeneration, run `target/sdf_coverage_diagnostic.py` and confirm:
   - All 10 restored cases show `sdf_min` ≤ -0.1 (inside-body samples present)
   - `sdf_negative_frac` ∈ [1e-4, 3e-4] (within bulk-train distribution)
   - 4D-Mahalanobis z-score < 2σ for all 10 cases
   - 32-bin chi² metric within bulk-train range

3. **Re-run baseline** — once the 10 SDFs are fixed, assign a clean baseline rerun to confirm the vol_p test improvement.

4. **Re-evaluate queued conditioning experiments** — several open PRs (#801, #802) are conditioning on SDF. They should be re-evaluated or re-run after the data fix lands.

## Acceptance Criterion

After regeneration, `target/sdf_coverage_diagnostic.py` should show all 10 restored cases falling within the bulk-train distribution on both 4D-Mahalanobis and 32-bin histogram chi² metrics (z < 2σ on both). The OOD-4 cluster in `target/analysis/sdf_pca.png` (currently left of PC1) should shift back into the main cloud.

This is the highest-leverage unblocked fix available. Will prioritize assigning a student to this immediately when a slot opens.
